### ISSUE TITLE
[BUG] Instagram stories

### ISSUE THREAD

**Issue by msalah-kw** [deniscerri/ytdlnis#225]
When I'm trying to download Instagram story from an account who has many stories, it downloads **only the first story**, however I choosed the second or the third story 

 - Android Version: 13
 - App Version: 1.6.3
 - Other Device Information: Samsung S22 ultra SM-S908E - Rooted with KernelSU

**Comment by deniscerri** [comment]
Please give me the link you used

**Comment by msalah-kw** [comment]
https://ume.la/BM5ze4

**Comment by msalah-kw** [comment]
Not only this one, any story from any account

**Comment by deniscerri** [comment]
![Screenshot_20230724-104410_YTDLnis](https://github.com/deniscerri/ytdlnis/assets/64997243/8c04d9ec-8981-45a8-a69d-a21030feadef)

Also when i put it in the search box it shows me all the videos. Can you show me a screen recording of what you are doing?

**Comment by msalah-kw** [comment]
Yes it shows all, but try to download anyone randomly except the first one
You will find that it downloaded the first story

**Comment by deniscerri** [comment]
I see. Ok noted

**Comment by deniscerri** [comment]
For now. Turn on quick download in the settings and try again. It won't show results but just the download card. Press download that's it

**Comment by msalah-kw** [comment]
I did that and it worked properly
Thank you bro for helping and for sharing this great app 👌🌷

**Comment by Fred-Vatin** [comment]
Hello,

This bug is still there on v1.8.4 using yt-dlp v2025.04.06.232826. Cookies are ok.

Go to any Instagram account that has multiple stories.

When pasting stories link in the home tab, the card and duration are ok but if I select one card that is not the first one and download, it will download the first one instead of the selected.

**Comment by deniscerri** [comment]
Can you send a sample link

**Comment by Fred-Vatin** [comment]
> Can you send a sample link

Every stories of any account as long as there are more than one story like here: https://www.instagram.com/stories/sarychevkirill/

**Comment by deniscerri** [comment]
@Fred-Vatin looking at the yt-dlp json dump, it seems yt-dlp is not providing correct webpage urls for each record.
It has the same url for all stories. That could be the reason why. You might need to forward this to yt-dlp's repo

Here is a json_dump reference with the command
```
yt-dlp --print "%(webpage_url)s" --flat-playlist "https://www.instagram.com/stories/sarychevkirill/" --cookies .\insta.txt
```

Output:

```
https://www.instagram.com/stories/sarychevkirill/
https://www.instagram.com/stories/sarychevkirill/
https://www.instagram.com/stories/sarychevkirill/
```

**Comment by deniscerri** [comment]
Also in the code i check for fields in this order whichever exists first in the json dump

```
jsonObject.getStringByAny("webpage_url", "original_url", "url")
```

If it has url tag and the app processed a playlist it will use it instead.

**Comment by deniscerri** [comment]
I guess if u have "Use Item URL instead of Playlist URL" enabled in the app turn it off and i guess it will work ok since it will use indexing with the playlist url. @Fred-Vatin

**Comment by Fred-Vatin** [comment]
Disabling "Use Item URL instead of Playlist URL" did the trick. Thanks.
